BLL/Services/DirectoryService.cs

```diff
@@ -32,15 +33,27 @@ public List<SpecializationDTO> GetAllSpecializations()
             return _mapper.Map<List<Specialization>, List<SpecializationDTO>>(_db.Specializations.GetAll().ToList());
         }
 
+        public void SaveSpecialization(List<SpecializationDTO> specializationsDto)
+        {
+            _db.Specializations.RemoveAll();
```

Comment:
it's not a good idea to remove all records
does any other table have references to Specialization? if yes, you cannot remove records from Specialization until you drop all references in the other tables